apps/web-connect/src/features/checkout/components/crossmint/FiatEmbeddedCheckoutIFrame.tsx

```diff
@@ -0,0 +1,35 @@
+import isEqual from "lodash.isequal";
+import {
+    type FiatEmbeddedCheckoutProps,
+    crossmintIFrameService,
+    embeddedCheckoutPropsToUpdatableParamsPayload,
+} from "@crossmint/client-sdk-base";
+
+import CrossmintEmbeddedCheckoutIFrame from "./EmbeddedCheckoutIFrame";
+import { type DependencyList, type EffectCallback, useEffect, useRef } from "react";
+
+function useDeepEffect(callback: EffectCallback, dependencies: DependencyList): void {
+    const dependenciesRef = useRef(dependencies);
+
+    useEffect(() => {
+        const hasChanged = dependencies.some((dep, i) => !isEqual(dep, dependenciesRef.current[i]));
+
+        if (hasChanged) {
+            dependenciesRef.current = dependencies;
+            return callback();
+        }
+    }, [dependencies]);
```

Comment:
Just FYI, I get a lint error here saying:

`React Hook useEffect has a missing dependency: 'callback'. Either include it or remove the dependency array. If 'callback' changes too often, find the parent component that defines it and wrap that definition in useCallback.`

Reply:
@XDapps Thats strange, what linter / config are you using. I do not get these warnings and they don't show up in the production build either https://github.com/xai-foundation/sentry/actions/runs/11395264488/job/31707162682?pr=459